src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -755,11 +755,7 @@ mesh::PtrData CouplingSchemeConfiguration::findDataByID(
     int ID) const
 {
   for (const mesh::PtrMesh &mesh : _meshConfig->meshes()) {
-    for (mesh::PtrData data : mesh->data()) {
-      if (data->getID() == ID) {
-        return data;
-      }
-    }
+    return mesh->data(ID);
```

Comment:
This requires ID to exist.

Reply:
Good point. Will do this.

Reply:
Done in 5ec6d2a761aeb1a5f6c81a65fde85f072eabe237